pkg/kubelet/kuberuntime/instrumented_services_test.go

```diff
@@ -22,26 +22,29 @@ import (
 	"testing"
 	"time"
 
+	"github.com/prometheus/client_golang/prometheus/promhttp"
 	"github.com/stretchr/testify/assert"
 
-	"k8s.io/component-base/metrics/legacyregistry"
+	compbasemetrics "k8s.io/component-base/metrics"
 	runtimeapi "k8s.io/cri-api/pkg/apis/runtime/v1alpha2"
 	"k8s.io/kubernetes/pkg/kubelet/metrics"
 )
 
 func TestRecordOperation(t *testing.T) {
-	legacyregistry.MustRegister(metrics.RuntimeOperations)
-	legacyregistry.MustRegister(metrics.RuntimeOperationsDuration)
-	legacyregistry.MustRegister(metrics.RuntimeOperationsErrors)
+	// Use local registry
+	var registry = compbasemetrics.NewKubeRegistry()
+	registry.MustRegister(metrics.RuntimeOperations)
+	registry.MustRegister(metrics.RuntimeOperationsDuration)
+	registry.MustRegister(metrics.RuntimeOperationsErrors)
 
 	l, err := net.Listen("tcp", "127.0.0.1:0")
 	assert.NoError(t, err)
 	defer l.Close()
 
 	prometheusURL := "http://" + l.Addr().String() + "/metrics"
 	mux := http.NewServeMux()
-	//lint:ignore SA1019 ignore deprecated warning until we move off of global registries
-	mux.Handle("/metrics", legacyregistry.Handler())
+	handler := promhttp.HandlerFor(registry, promhttp.HandlerOpts{})
```

Comment:
You can't import from prometheus directly, you'll need to use one of our [wrapped handlers](https://github.com/kubernetes/kubernetes/blob/ea0764452222146c47ec826977f49d7001b0ea8c/staging/src/k8s.io/component-base/metrics/http.go#L62-L64).

Reply:
`Defer` inside for loop is a little tricky. 

It will begin to`Reset` after all loops finish instead of one loop finishes.
https://play.golang.org/p/9e9qyuE64nI

Reply:
It worked. Thanks!